ocaml/xapi-cli-server/cli_util.ml

```diff
@@ -165,10 +165,11 @@ let track_http_operation ?use_existing_task ?(progress_bar = false) fd rpc
           debug "result was [%s]" result ;
           result
         ) else
-          let params =
-            Client.Task.get_error_info ~rpc ~session_id ~self:task_id
-          in
-          raise (Api_errors.Server_error (List.hd params, List.tl params))
+          match Client.Task.get_error_info ~rpc ~session_id ~self:task_id with
+          | [] ->
+              raise Api_errors.(Server_error (internal_error, []))
```

Comment:
I think it's worth defining a function `internal_error` for this related pattern. @last-genius recently defined one, maybe it's accessible from here or could be added.